### DISCUSSION TITLE
Does v2 have the same way of accessing the themeConfig from a Layout.vue that v1 does?

### DISCUSSION THREAD

**Discussion by ArsalaBangash** [vuepress/core#686] [Q&A] [answered]
Hi, Vuepress community! 

### Context

I'm experimenting with making a vuepress2 version of [vuepress-theme-casper](https://github.com/alexander-heimbuch/vuepress-theme-casper]. 

Current progress is over [here](https://gitlab.com/grey-software/research/vuepress2-theme-ghost-casper)

### Q: Does v2 have the same way of accessing the themeConfig that v1 does?

[This is where the author sends `this.$site` to vuex](https://github.com/alexander-heimbuch/vuepress-theme-casper/blob/e2f02b4b1c8c6d3573dba9a2680ad411bd5d3012/blog/.vuepress/theme/Layout.vue#L58)

[This is where the author accesses `site.themeConfig` from vuex](https://github.com/alexander-heimbuch/vuepress-theme-casper/blob/e2f02b4b1c8c6d3573dba9a2680ad411bd5d3012/blog/.vuepress/theme/store/mutations.js#L8)

When I've been trying this on the [vuepress2-theme-ghost-casper repo](https://gitlab.com/grey-software/research/vuepress2-theme-ghost-casper), I can't seem to locate where the themeConfig object is stored so vuex can interact with it.

**Comment by Mister-Hope** [comment, marked as answer]
Hey, plz read the docs, we highly recommand you to use composition api.

You can use `@vuepress/plugin-theme-data` and it's `useThemeData` compoition api to get a reactive themeData.

`@vuepress/theme-default` can be a demo.

**Reply by ArsalaBangash** [reply]
Thank you for the quick answer! I totally missed this. Maybe this should be mentioned in the migration documentation and I am open to submitting a pull request for it.

**Reply by Mister-Hope** [reply]
I agree that it would be nice to have this changes mentioned in docs. Welcome for a pr adding some description here: https://v2.vuepress.vuejs.org/guide/migration.html#for-theme-authors

**Reply by Mister-Hope** [reply]
Also I would be glad helping you to complete chinese docs after you finished the english ones.
